### ISSUE TITLE
use push_navigate in nested liveview causes full reload

### ISSUE THREAD

**Issue by limeuser** [phoenixframework/phoenix_live_view#3424]
### Environment

* Elixir version (elixir -v): 1.17.2
* Phoenix version (mix deps): 1.7.14
* Phoenix LiveView version (mix deps): 1.0.0-rc.1
* Operating system: ubuntu 20.04
* Browsers you attempted to reproduce this bug on (the more the merrier): 
* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no: Yes

### Actual behavior
In my application, i use nested liveview in layout to implement sidebar. In sidevar liveview, using <.link navigate> navgite to liveview is ok, but using push_navigate in click handle_event cause full reload, and liveview socket is reconnected.

#### app.html.heex   implement sidebar as nested liveview in layout
```elixir
<div>
  <%= live_render(@socket, TestliveWeb.SideBarLive, sticky: true, id: :side_bar) %>
  <%= @inner_content %>
</div>
```

#### side_bar_live.ex
* navigate by <.link> is ok
* navigate by click event, call push_naviagte in handle_event callback cause liveview reconnect
```elixir
defmodule TestliveWeb.SideBarLive do
  use TestliveWeb, :live_view

  def render(assigns) do
    ~H"""
    <div class="flex flex-col">
      <span>sidebar</span>
      <.link navigate={~p"/articles/2/show"}>article2_link_navigate</.link>
      <a phx-click="nav_to_article" phx-value-id="2">article2_push_navigate</a>
    </div>
    """
  end

  def handle_event("nav_to_article", %{"id" => id}, socket) do
    {:noreply, socket |> push_navigate(to: ~p"/articles/#{id}/show")}
  end
end
```

#### article_live.ex
```
defmodule TestliveWeb.ArticleLive do
  use TestliveWeb, :live_view

  @impl true
  def render(assigns) do
    ~H"""
    <div>
      <%= @id %>
    </div>
    """
  end

  @impl true
  def mount(%{"id" => id}, _session, socket) do
    {:ok, assign(socket, id: id)}
  end
end
```

#### main_live.ex
```elixir
defmodule TestliveWeb.MainLive do
  use TestliveWeb, :live_view

  @impl true
  def render(assigns) do
    ~H"""
    <div>
      <.link navigate={~p"/articles/1/show"}>article1</.link>
    </div>
    """
  end
end
```

#### router
```elixir
 scope "/", TestliveWeb do
    pipe_through :browser
    live "/main", MainLive, :index
    live "/articles/:id/show", ArticleLive
  end
```

### Expected behavior
* use push_naviagte in nested liveview don't causes full reload

**Comment by SteffenDE** [comment]
This is fixed in https://github.com/phoenixframework/phoenix_live_view/pull/3280.

cc @chrismccord

**Comment by chrismccord** [comment]
Closed via #3470
